Change debian/* to build 5.6.34

holser opened the pull request:
- Bumped version numbers
- added systemd service profile and script
- Standards version moved up to 3.9.6 in d/control
- Added dependencies to gawk as one script does not work with original-awk
- Dropped explicit dependency on multiarch-support, this is already managed
  by ${misc:Pre-Depends).
- Added "Provides: libmysqlclient-dev" to mysql-wsrep-libmysqlclient-dev
- Added "Replaces: libmysqlclient-dev" to mysql-wsrep-libmysqlclient-dev
- Moved mysql_plugin to server package as it controls the server plugins
- Removed langueage dublication from d/mysql-wsrep-server-5.6.install
- Make builds parallel Use MAKEFLAGS for parallel builds
- Added parallel build support via DEB_BUILD_OPTIONS
- d/rules: call dh_apparmor from override_dh_install so that it is run
  before dh_installinit. This causes the AppArmor profile to be loaded
  correctly before mysqld is started.
- Added innodb_stress tests to d/rules
-Replace testsuite dependency on python with libjson-perl
-Remove unittest/unit.pl from d/rules
  Unit tests are not run with ctest instead of perl, so this file has
  been removed from the source

mmalchuk commented:
+1

ottok commented:
Good work!

Some questions:
- Why do you remove my.cnf?
- Why did you move usr/local to the last in the PATH? Now ccache that wraps gcc will not pick up before real gcc is called.
- Building with --with-systemd is not backwards compatible. These packages will stop building on anything older than Debian Jessie or Ubuntu Xenial.

holser commented:
Hi Otto,

> Why do you remove my.cnf?

I pushed it back. Now, it should create less confusion

> Why did you move usr/local to the last in the PATH? Now ccache that wraps gcc will not pick up before real gcc is called.

I moved the order back to original.

> Building with --with-systemd is not backwards compatible. These packages will stop building on anything older than Debian Jessie or Ubuntu Xenial.

It's fully compatible with Ubuntu Trusty as it was tested. Please have a look at debian/rules line 260-263 (dh_systemd_enable)

Just to give a highlight. This PR is based on [ debian package ](https://anonscm.debian.org/git/pkg-mysql/mysql-5.6.git) commit fcd2771abdb7cc4809f6dfce6d900dbf612e5113

Also, I've updated it to include a few changes Debian maintainers made to keep the difference as little as possible.
